tests/pages/api/multipleProductValidity.test.ts

```diff
@@ -1,169 +1,68 @@
-                productNameId: '',
-                productPrice: '30.90',
-                productPriceId: '',
-                productDuration: '30',
-                productDurationId: '',
-                productValidity: 'endOfServiceDay',
-            };
-            const result = addErrorsIfInvalid(req, product, userInputIndex);
-
-            expect(result.productValidity).toBe('endOfServiceDay');
-            expect(result.productValidityError).toBe('Specify an end time for service day');
-        });
-
-        it('add error when validity end time is entered incorrectly', () => {
-            const { req } = getMockRequestAndResponse({
-                body: {
-                    'validity-option-0': 'endOfServiceDay',
-                    'validity-end-time-0': '2400',
-                },
-            });
-
-            const userInputIndex = 0;
-            const product: Product = {
-                productName: 'best ticket',
-                productNameId: '',
-                productPrice: '30.90',
-                productPriceId: '',
-                productDuration: '30',
-                productDurationId: '',
-                productValidity: 'endOfServiceDay',
-            };
-            const result = addErrorsIfInvalid(req, product, userInputIndex);
-
-            expect(result.productValidity).toBe('endOfServiceDay');
-            expect(result.productValidityError).toBe('2400 is not a valid input. Use 0000.');
+    it('updates the MULTIPLE_PRODUCT_ATTRIBUTE and redirects to selectSalesOfferPackage page if all valid', () => {
+        const { req, res } = getMockRequestAndResponse({
+            cookieValues: { fareZoneName: null },
+            body: {
+                'validity-option-0': '24hr',
+                'validity-option-1': '24hr',
+                'validity-option-2': 'endOfCalendarDay',
+            },
+            mockWriteHeadFn: writeHeadMock,
         });
+        multipleProductValidity(req, res);
 
-        it('should not error if there is whitespace in the time', () => {
-            const { req } = getMockRequestAndResponse({
-                body: {
-                    'validity-option-0': 'endOfServiceDay',
-                    'validity-end-time-0': ' 1200',
-                },
-            });
-
-            const userInputIndex = 0;
-            const product: Product = {
-                productName: 'best ticket',
-                productNameId: '',
-                productPrice: '30.90',
-                productPriceId: '',
-                productDuration: '30',
-                productDurationId: '',
-                productValidity: 'endOfServiceDay',
-            };
-            const result = addErrorsIfInvalid(req, product, userInputIndex);
-
-            expect(result.productValidity).toBe('endOfServiceDay');
-            expect(result.productValidityError).toBe(undefined);
+        expect(writeHeadMock).toBeCalledWith(302, {
+            Location: '/ticketConfirmation',
```

Comment:
Why is this location different to the redirect expected according to this test's 'it' description?

Reply:
good catch